### PR TITLE
Connect to QScreen for orientation change updates

### PR DESCRIPTION
**Pull request by bzeller**

### CODE REVIEW

**Comment by faenil** [comment]
can you squash the second commit?

maybe @rburchell can review this ;)

looks okay to me though

File: src/meego/mdeclarativescreen.cpp [hunk 1]
```diff
@@ -255,6 +254,17 @@ void MDeclarativeScreenPrivate::initContextSubscriber()
                      q, SLOT(_q_updateOrientationAngle()));
     QObject::connect(&remoteTopEdgeListener, SIGNAL(nameDisappeared()),
                      q, SLOT(_q_updateOrientationAngle()));
+
+#elif  QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+    QScreen* screen = QGuiApplication::primaryScreen();
```
Comment (rburchell) [inline src/meego/mdeclarativescreen.cpp:228]: - goes with the screen

File: src/meego/mdeclarativescreen.cpp [hunk 2]
```diff
@@ -255,6 +254,17 @@ void MDeclarativeScreenPrivate::initContextSubscriber()
...
+    if (window)
+        screen = window.data()->screen();
+    if (screen) {
+        QObject::connect(screen, SIGNAL(orientationChanged(Qt::ScreenOrientation)),
+                         q, SLOT(_q_updateOrientationAngle()));
+    } else {
+        qWarning()<<"No valid QScreen found, rotation will not work!";
```
Comment (rburchell) [inline src/meego/mdeclarativescreen.cpp]: space after ) and before "

**Comment by rburchell** [comment]
generally looks ok. please also drop the qt4-specific code, if you've some spare time -- this branch is qt5-only (even though it's still littered with conditionals)

File: src/meego/mdeclarativeinputcontext.cpp [hunk 3]
```diff
@@ -66,9 +62,8 @@ class MDeclarativeInputContextPrivate
     MDeclarativeInputContextPrivate(MDeclarativeInputContext *qq);
     ~MDeclarativeInputContextPrivate();
 
-#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+
```
Comment (sledges) [inline src/meego/mdeclarativeinputcontext.cpp]: Extra newline (just to tell you I reviewed it :D)

File: src/meego/mdeclarativeinputcontext.cpp [hunk 4]
```diff
@@ -280,18 +243,11 @@ QQuickItem *MDeclarativeInputContext::customSoftwareInputPanelTextField() const
     return d->sipVkbTextField;
 }
 
-#if QT_VERSION >= 0x050000
-void MDeclarativeInputContext::setCustomSoftwareInputPanelTextField(QQuickItem *item)
-#else
+
```
Comment (sledges) [inline src/meego/mdeclarativeinputcontext.cpp]: Extra newline. And oddly this #ifdef had been totally redundant :)

File: src/meego/mdeclarativeinputcontext.cpp [hunk 5]
```diff
@@ -322,19 +278,10 @@ void MDeclarativeInputContext::simulateSipClose()
...
-    if (ic) {
-        Maliit::PreeditInjectionEvent event(newPreedit, eventCursorPosition);
-        event.setReplacement(replacementStart, replacementLength);
-        return ic->filterEvent(&event);
-    }
-#else
     Q_UNUSED(newPreedit)
     Q_UNUSED(eventCursorPosition)
     Q_UNUSED(replacementStart)
     Q_UNUSED(replacementLength)
```
Comment (sledges) [inline src/meego/mdeclarativeinputcontext.cpp:283]: Please add // FIXME: How do we deal with this?

File: src/meego/mdeclarativemousefilter.cpp [hunk 6]
```diff
@@ -185,11 +170,8 @@ bool MDeclarativeMouseFilter::sceneEvent(QEvent *event)
             emit finished();
             return true;            
         }
-#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+
```
Comment (sledges) [inline src/meego/mdeclarativemousefilter.cpp]: Extra newline

File: src/meego/mdeclarativescreen.cpp [hunk 7]
```diff
@@ -380,19 +344,35 @@ int MDeclarativeScreenPrivate::rotation() const
...
-
-    if (topEdge == "top") {
-        o = MDeclarativeScreen::Landscape;
-    } else if (topEdge == "left") {
-        o = MDeclarativeScreen::Portrait;
-    } else if (topEdge == "right") {
-        o = MDeclarativeScreen::PortraitInverted;
-    } else if (topEdge == "bottom") {
-        o = MDeclarativeScreen::LandscapeInverted;
+
```
Comment (sledges) [inline src/meego/mdeclarativescreen.cpp]: Extra newline

File: src/meego/mdeclarativescreen.cpp [hunk 8]
```diff
@@ -380,19 +344,35 @@ int MDeclarativeScreenPrivate::rotation() const
...
+            }
+        case Qt::InvertedPortraitOrientation: {
+                o = MDeclarativeScreen::PortraitInverted;
+                break;
+            }
+        case Qt::PrimaryOrientation: break;
+        }
     }
-#endif
+
```
Comment (sledges) [inline src/meego/mdeclarativescreen.cpp]: Extra newline (sorry for nitpicking :))

File: src/meego/mdeclarativescreen.cpp [hunk 9]
```diff
@@ -779,31 +719,12 @@ int MDeclarativeScreen::dpi() const {
         return ExtraHigh;
 }
 
-#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
-    void MDeclarativeScreen::updatePlatformStatusBarRect(QQuickItem * statusBar)
-#else
-    void MDeclarativeScreen::updatePlatformStatusBarRect(QDeclarativeItem * statusBar)
-#endif
+
```
Comment (sledges) [inline src/meego/mdeclarativescreen.cpp]: Extra newline

File: src/meego/minversemousearea.cpp [hunk 10]
```diff
@@ -77,11 +77,8 @@ void MInverseMouseArea::setEnabled(bool enabled)
     }
 }
 
-#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+
```
Comment (sledges) [inline src/meego/minversemousearea.cpp]: Extra newline

**Comment by sledges** [comment]
LGTM after cosmetics

File: src/meego/mdeclarativemousefilter.cpp (hunk 6 above)
Comment (bzeller) [inline src/meego/mdeclarativemousefilter.cpp]: This newline makes sense IMHO. Removing it looks wrong ;)
But ok ;).

**Pull request merged**